libraries/AP_RCProtocol/AP_RCProtocol.h

```diff
@@ -197,28 +186,33 @@ class AP_RCProtocol {
     // return true if a specific protocol is enabled
     bool protocol_enabled(enum rcprotocol_t protocol) const;
 
+    void add_input(enum rcprotocol_t protocol, uint8_t num_channels, uint16_t *values, int16_t rssi, int16_t rx_link_quality);
+
     enum rcprotocol_t _detected_protocol = NONE;
     uint16_t _disabled_for_pulses;
     bool _detected_with_bytes;
     AP_RCProtocol_Backend *backend[NONE];
     bool _new_input;
-    uint32_t _last_input_ms;
+    uint32_t _last_input_us;
     bool _failsafe_active;
     bool _valid_serial_prot;
 
     // optional additional uart
     struct {
+        SoftSerial ss;
```

Comment:
Why is this in the "added" structure?

Reply:
Yes I wondered about that. Can move it out.

Reply:
Fixed